controllers/mcad_controller_test.go

```diff
@@ -0,0 +1,55 @@
+package controllers
+
+import (
+	"context"
+	mfc "github.com/manifestival/controller-runtime-client"
+	mf "github.com/manifestival/manifestival"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	codeflarev1alpha1 "github.com/project-codeflare/codeflare-operator/api/v1alpha1"
+	"github.com/project-codeflare/codeflare-operator/controllers/testutil"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	mcadCRCase1    = "./testdata/mcad_test_cases/case_1.yaml"
+	mcadConfigMap1 = "./testdata/mcad_test_results/case_1/configmap.yaml"
+)
+
+func deployMCAD(ctx context.Context, path string, opts mf.Option) {
+	mcad := &codeflarev1alpha1.MCAD{}
+	err := convertToStructuredResource(path, mcad, opts)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient.Create(ctx, mcad)).Should(Succeed())
+}
```

Comment:
Does OpenShift CI deploy a cluster when running tests? This is definitely more of an integration test unless I'm missing a mock somewhere.

Reply:
This is actually mocking the api, not deploying real objects on an actual cluster.

Reply:
You can try running the test yourself via `make test-unit` on an environment that isn't logged in on any OpenShift/k8s cluster.